pkg/deploy/reaper/reaper.go

```diff
@@ -65,6 +66,20 @@ func (reaper *DeploymentConfigReaper) Stop(namespace, name string, timeout time.
 			// Better not error out here...
 			glog.Infof("Cannot delete ReplicationController %s/%s: %v", rc.Namespace, rc.Name, err)
 		}
+
+		// Delete all deployer and hook pods
```

Comment:
You don't need to try and list for all deployments. Since the deployment controller already deletes the deployers for Complete and Cancelled deployments, you only need to delete deployers for Failed non-cancelled deployments.